File: Lab 1/src/main/java/uk/ac/alc/wpd2/callumcarmicheal/messageboard/web/ViewUtil.java
package uk.ac.alc.wpd2.callumcarmicheal.messageboard.web;

import com.google.common.primitives.Ints;

import static org.apache.commons.lang3.StringEscapeUtils.escapeHtml4;

public class ViewUtil {

    public String Escape(String e) {
        return escapeHtml4(e);
    }
    public int I(Long l) {return Ints.checkedCast(l);}
}

File: Kotlin_Test/src/main/kotlin/com/callumcarmicheal/wframe/ViewUtil.kt
package com.callumcarmicheal.wframe

import com.google.common.primitives.Ints
import org.unbescape.html.HtmlEscape

class ViewUtil {
    fun E(e: String): String {
        return HtmlEscape.escapeHtml5(e)
    }

    fun UE(e: String): String {
        return HtmlEscape.unescapeHtml(e)
    }

    fun I(l: Long): Int {
        return Ints.checkedCast(l)
    }

    fun GenerateColorFromString(x: String): String {
        val i = x.hashCode()
        val c = Integer.toHexString(i and 0x00FFFFFF)
            .toUpperCase()

        return "00000".substring(0, 6 - c.length) + c
    }
}